MapboxNavigation/NavigationMapView.swift

```diff
@@ -6,12 +6,13 @@ import Turf
 
 typealias CongestionSegment = ([CLLocationCoordinate2D], CongestionLevel)
 
-let routeLineWidthAtZoomLevels: [Int: MGLStyleValue<NSNumber>] = [
-    10: MGLStyleValue(rawValue: 8),
-    13: MGLStyleValue(rawValue: 9),
-    16: MGLStyleValue(rawValue: 11),
-    19: MGLStyleValue(rawValue: 22),
-    22: MGLStyleValue(rawValue: 28)
+
+let routeLineWidthAtZoomLevels: [Int: NSExpression] = [
+    10: NSExpression(forConstantValue: 8),
+    13: NSExpression(forConstantValue: 9),
+    16: NSExpression(forConstantValue: 11),
+    19: NSExpression(forConstantValue: 22),
+    22: NSExpression(forConstantValue: 28)
```

Comment:
@anandthakker, currently, it isn’t possible for a zoom expression (`$zoomLevel`) to appear outside a curve or interpolation. But would it be feasible to relax this restriction, so that this curve could be expressed as a single multiplication expression involving `$zoomLevel`?

Reply:
Technically, it would be _possible_ to relax the restriction on zoom expressions to:
 - If the expression is a zoom-only expression, no restrictions at all.
 - If the expression is also feature-dependent, it must be in the form `F(zoom) * G(feature)`, where `F` is a step or interpolation expression using `zoom` as the input (and nowhere else), and `G` is not zoom-dependent.

We've chosen so far to go with the existing, less permissive restriction because it's so much simpler to understand.  I think the above, which is maximally permissive (barring significant changes to the data-driven styling implementation), is likely way too complicated.  We could go for something in between, but it'll be tricky to decide just where to draw the line.